### PR TITLE
samples: tests: Add support to tests on nrf54l15 DK

### PR DESCRIPTION
**Pull request by katgiadla**
Align all existing samples/tests/applications, which contains `nrf54l15pdk/nrf54l15/*` by adding
`nrf54l15dk/nrf54l15/*` to enable twister builds.

### CODE REVIEW

**Review by PerMac — CHANGES_REQUESTED** [review]
please, remove pdk from integration_platforms whenever you add dk. It seems many of integration_platforms in these samples are already overloaded so at least we won't expand it even further. Those lists still need clearing, but it would a topic for another PR

File: samples/subsys/ipc/ipc_service/icmsg/boards/nrf54l15dk_nrf54l15_cpuapp.overlay [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+			};
+		};
+	};
+
+	ipc {
+		ipc0: ipc0 {
+			compatible = "zephyr,ipc-icmsg";
+			tx-region = <&sram_tx>;
+			rx-region = <&sram_rx>;
+			mboxes = <&cpuapp_vevif_rx 15>, <&cpuapp_vevif_tx 16>;
```
Comment (jaz1-nordic) [inline samples/subsys/ipc/ipc_service/icmsg/boards/nrf54l15dk_nrf54l15_cpuapp.overlay:28]:
```suggestion
			mboxes = <&cpuapp_vevif_rx 20>, <&cpuapp_vevif_tx 21>;
```

File: samples/subsys/ipc/ipc_service/icmsg/remote/boards/nrf54l15dk_nrf54l15_cpuflpr.overlay [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+			};
+		};
+	};
+
+	ipc {
+		ipc0: ipc0 {
+			compatible = "zephyr,ipc-icmsg";
+			tx-region = <&sram_tx>;
+			rx-region = <&sram_rx>;
+			mboxes = <&cpuflpr_vevif_rx 16>, <&cpuflpr_vevif_tx 15>;
```
Comment (jaz1-nordic) [inline samples/subsys/ipc/ipc_service/icmsg/remote/boards/nrf54l15dk_nrf54l15_cpuflpr.overlay:28]:
```suggestion
			mboxes = <&cpuflpr_vevif_rx 21>, <&cpuflpr_vevif_tx 20>;
```

File: samples/drivers/mbox/remote/boards/nrf54l15dk_nrf54l15_cpuflpr_xip.overlay [hunk 3]
```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/ {
+	mbox-consumer {
+		compatible = "vnd,mbox-consumer";
+		mboxes = <&cpuflpr_vevif_rx 16>, <&cpuflpr_vevif_tx 15>;
```
Comment (jaz1-nordic) [inline samples/drivers/mbox/remote/boards/nrf54l15dk_nrf54l15_cpuflpr_xip.overlay:9]:
```suggestion
		mboxes = <&cpuflpr_vevif_rx 21>, <&cpuflpr_vevif_tx 20>;
```

File: tests/drivers/mbox/mbox_error_cases/boards/nrf54l15dk_nrf54l15_cpuapp.overlay [hunk 4]
```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/ {
+	mbox-consumer {
+		compatible = "vnd,mbox-consumer";
+		mboxes = <&cpuapp_vevif_tx 16>, <&cpuapp_vevif_tx 32>,
```
Comment (jaz1-nordic) [inline tests/drivers/mbox/mbox_error_cases/boards/nrf54l15dk_nrf54l15_cpuapp.overlay:9]:
```suggestion
		mboxes = <&cpuapp_vevif_tx 21>, <&cpuapp_vevif_tx 32>,
```

File: tests/drivers/mbox/mbox_error_cases/boards/nrf54l15dk_nrf54l15_cpuapp.overlay [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+				 <&cpuapp_vevif_rx 15>, <&cpuapp_vevif_rx 32>;
```
Comment (jaz1-nordic) [inline tests/drivers/mbox/mbox_error_cases/boards/nrf54l15dk_nrf54l15_cpuapp.overlay:10]:
```suggestion
				 <&cpuapp_vevif_rx 20>, <&cpuapp_vevif_rx 32>;
```

**Review by jaz1-nordic — CHANGES_REQUESTED** [review]

File: samples/subsys/ipc/ipc_service/icmsg/boards/nrf54l15dk_nrf54l15_cpuapp.overlay (hunk 1 above)
Comment (katgiadla) [reply samples/subsys/ipc/ipc_service/icmsg/boards/nrf54l15dk_nrf54l15_cpuapp.overlay]: done

File: samples/subsys/ipc/ipc_service/icmsg/remote/boards/nrf54l15dk_nrf54l15_cpuflpr.overlay (hunk 2 above)
Comment (katgiadla) [reply samples/subsys/ipc/ipc_service/icmsg/remote/boards/nrf54l15dk_nrf54l15_cpuflpr.overlay]: done

File: samples/drivers/mbox/remote/boards/nrf54l15dk_nrf54l15_cpuflpr_xip.overlay (hunk 3 above)
Comment (katgiadla) [reply samples/drivers/mbox/remote/boards/nrf54l15dk_nrf54l15_cpuflpr_xip.overlay]: done

File: tests/drivers/mbox/mbox_error_cases/boards/nrf54l15dk_nrf54l15_cpuapp.overlay (hunk 4 above)
Comment (katgiadla) [reply tests/drivers/mbox/mbox_error_cases/boards/nrf54l15dk_nrf54l15_cpuapp.overlay]: done

File: tests/drivers/mbox/mbox_error_cases/boards/nrf54l15dk_nrf54l15_cpuapp.overlay (hunk 5 above)
Comment (katgiadla) [reply tests/drivers/mbox/mbox_error_cases/boards/nrf54l15dk_nrf54l15_cpuapp.overlay]: done

File: samples/boards/nrf/system_off/sample.yaml [hunk 6]
```diff
@@ -25,6 +26,7 @@ tests:
     extra_args: DTC_OVERLAY_FILE="boards/nrf54l15pdk_nrf54l15_cpuapp_retained_mem.overlay"
```
Comment (nordic-piks) [inline samples/boards/nrf/system_off/sample.yaml:26]: nrf54l15pdk_nrf54l15_cpuapp_retained_mem->nrf54l15dk_nrf54l15_cpuapp_retained_mem

**Review by PerMac — DISMISSED** [review]

File: tests/drivers/uart/uart_elementary/testcase.yaml [hunk 7]
```diff
@@ -27,33 +29,63 @@ tests:
     extra_configs:
       - CONFIG_DUAL_UART_TEST=y
       - CONFIG_SETUP_MISMATCH_TEST=y
-  drivers.uart.uart_elementary_dual_nrf54l:
+  drivers.uart.uart_elementary_dual_nrf54lpdk:
     filter: CONFIG_SERIAL_SUPPORT_INTERRUPT
     platform_allow:
       - nrf54l15pdk/nrf54l15/cpuapp
     extra_args: DTC_OVERLAY_FILE="boards/nrf54l15pdk_nrf54l15_cpuapp_dual_uart.overlay"
```
Comment (nordic-piks) [inline tests/drivers/uart/uart_elementary/testcase.yaml:36]: pdk->dk

File: samples/boards/nrf/system_off/sample.yaml (hunk 6 above)
Comment (katgiadla) [reply samples/boards/nrf/system_off/sample.yaml]: done

File: tests/drivers/uart/uart_elementary/testcase.yaml (hunk 7 above)
Comment (katgiadla) [reply tests/drivers/uart/uart_elementary/testcase.yaml]: done

**Review by nordicjm — DISMISSED** [review]

File: samples/boards/nrf/system_off/sample.yaml [hunk 8]
```diff
@@ -22,9 +23,10 @@ tests:
     extra_configs:
       - CONFIG_APP_USE_NRF_RETENTION=y
   sample.boards.nrf.system_off.retained_mem:
-    extra_args: DTC_OVERLAY_FILE="boards/nrf54l15pdk_nrf54l15_cpuapp_retained_mem.overlay"
+    extra_args: DTC_OVERLAY_FILE="boards/nrf54l15dk_nrf54l15_cpuapp_retained_mem.overlay"
     build_only: true
     platform_allow:
+      - nrf54l15dk/nrf54l15/cpuapp
       - nrf54l15pdk/nrf54l15/cpuapp
```
Comment (magp-nordic) [inline samples/boards/nrf/system_off/sample.yaml:30]: Would't that cause using overlay for DK when building for PDK with twister? Maybe PDK should be removed here from `platform_allow` as in UARTE tests?

**Review by magp-nordic — DISMISSED** [review]
LGTM, just one note

File: samples/boards/nrf/system_off/sample.yaml (hunk 8 above)
Comment (katgiadla) [reply samples/boards/nrf/system_off/sample.yaml]: In this case, I could resign from support PDK. 
@nordic-piks, do you agree with them?

File: samples/bluetooth/hci_uart/boards/nrf54l15dk_nrf54l15_cpuapp.overlay [hunk 9]
```diff
@@ -4,6 +4,9 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-&wdt31 {
+&uart20 {
```
Comment (nordic-piks) [inline samples/bluetooth/hci_uart/boards/nrf54l15dk_nrf54l15_cpuapp.overlay:7]: ??

Comment (katgiadla) [reply samples/bluetooth/hci_uart/boards/nrf54l15dk_nrf54l15_cpuapp.overlay:7]: Rename of the file does not change the file content.

**Review by nordic-piks — APPROVED** [review]

**Review by PerMac — DISMISSED** [review]

**Review by magp-nordic — APPROVED** [review]

**Review by jaz1-nordic — APPROVED** [review]

**Comment by PerMac** [comment]
closed in favor of #78888

**Pull request closed without merging**